gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineNullAway.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.plugins;
+
+import com.google.common.collect.ImmutableSet;
+import java.util.Optional;
+import net.ltgt.gradle.errorprone.ErrorProneOptions;
+import org.gradle.api.Action;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.api.plugins.ExtensionAware;
+import org.gradle.api.tasks.compile.JavaCompile;
+
+public final class BaselineNullAway implements Plugin<Project> {
+
+    private static final Logger log = Logging.getLogger(BaselineNullAway.class);
+
+    private static final ImmutableSet<String> DEFAULT_ANNOTATED_PACKAGES = ImmutableSet.of(
+            "com.palantir",
+            // guava
+            "com.google.common");
+
+    @Override
+    public void apply(Project project) {
+        project.getPluginManager().withPlugin("com.palantir.baseline-error-prone", _unused0 -> {
+            project.getPluginManager().withPlugin("java", _unused1 -> {
+                applyToProject(project);
+            });
+        });
+    }
+
+    private void applyToProject(Project project) {
+        String version = Optional.ofNullable(BaselineNullAway.class.getPackage().getImplementationVersion())
+                .orElseGet(() -> {
+                    log.warn("BaselineNullAway is using 'latest.release' - "
+                            + "beware this compromises build reproducibility");
+                    return "latest.release";
```

Comment:
When do this ever happen? How is nullaway applied?

Reply:
This is the same logic used in baseline-error-prone. This happens when running integration tests from source, when the classpath is built up of class files rather than a jar with an `Implementation-Version` in the manifest.

Reply:
Do most jars like google's own error prone core jar have this manifest usually, or is this something we publish?

Reply:
No, this is used so that we can publishToMavenLocal prior prior to running tests, then resolve our own checks from the `mavenLocal()` repo to test them.